plugins/kiro/scripts/heartbeat.sh

```
@@ -0,0 +1,165 @@
+HOST="unknown"         # hostname; both set by resolve_version via install-id.sh
+TRIGGER="SessionStart" # which hook fired us; anything else is rate-limited
+
+# Self-locate the plugin root from $0 (<root>/scripts/heartbeat.sh).
+locate_plugin_root() {
+  PLUGIN_ROOT="$(CDPATH= cd -- "$(dirname -- "$0")/.." 2>/dev/null && pwd)"
+  [ -n "$PLUGIN_ROOT" ] || PLUGIN_ROOT="${KIRO_PLUGIN_ROOT:-.}"
+}
+
+# Same env precedence as hook.sh (later wins): bundled → MDM → per-user.
+load_env() {
+  [ -r "${PLUGIN_ROOT}/env" ] && . "${PLUGIN_ROOT}/env"
```

Shellcheck (0.11.0)

[info] Not following: ./env was not specified as input (see shellcheck -x).

(SC1091)